composer/workflows/dataflowtemplateoperator_tutorial.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START composer_dataflow_dag]
+
+
+# Example Airflow DAG that creates a Cloud Dataproc cluster, runs the Hadoop
+# wordcount example, and deletes the cluster.
+
+# This DAG relies on three Airflow variables
+# https://airflow.apache.org/concepts.html#variables
+# * project_id - Google Cloud Project ID to use for the Cloud Dataflow cluster.
+# * gce_zone - Google Compute Engine zone where Cloud Dataflow cluster should be
+#   created.
+# * gce_region - Google Compute Engine region where Cloud Dataflow cluster should be
+#   created.
+# Learn more about the difference between the two here:
+# https://cloud.google.com/compute/docs/regions-zones
+# * gcs_bucket - Google Cloud Storage bucket where you've stored the User Defined 
+# Function (.js), the input file (.txt), and the JSON schema (.json).
+
+
+import datetime
+
+from airflow.utils.dates import days_ago
+from airflow.contrib.operators.dataflow_operator import DataflowTemplateOperator
+from airflow import models
+
+bucket_path = "gs://" + models.Variable.get("bucket_name")
+project_id = models.Variable.get("project_id")
+gce_zone = models.Variable.get("gce_zone")
+gce_region = models.Variable.get("gce_region")
+
+
+default_args = {
+    # Tell airflow to start one day ago, so that it runs as soon as you upload it
+    "start_date": days_ago(1),
+    "dataflow_default_options": {
+        "project": project_id,
+        # Set to your region
+        "region": gce_region,
+        # Set to your zone
+        "zone": gce_zone,
+        # This is a subfolder for storing temporary files, like the staged pipeline job.
+        "temp_location": bucket_path + "/tmp/",
+    },
+}
+
+# Define a DAG (directed acyclic graph) of tasks.
+# Any task you create within the context manager is automatically added to the
+# DAG object.
+with models.DAG(
+    # The id you will see in the DAG airflow page
+    "composer_dataflow_dag",
+    default_args=default_args,
+    # The interval with which to schedule the DAG
+    schedule_interval=datetime.timedelta(days=1),  # Override to match your needs
+) as dag:
+
+    start_template_job = DataflowTemplateOperator(
+        # The task id of your job
+        task_id="dataflow_operator_transform_CSV_to_BQ",
```

Comment:
my DAG failed with this task ID unfortunately - there can't be any capital letters in the task_id (dumb airflow requirement)